tests/oeedger8r/enc/testpointer.cpp

```diff
@@ -266,7 +266,17 @@ static T* ecall_pointer_fun_impl(
         if (p4)
         {
             OE_TEST(oe_is_within_enclave(p4, sizeof(T) * 16));
-            OE_TEST(memcmp(p4, exp, sizeof(T) * 16) == 0);
+
```

Comment:
Nice explanation.

Reply:
Great work John.
@anakrish - Is there a way we can avoid duplication of the set of compiler flags i.e. have clangw read this from compiler_settings.cmake?

Reply:
> 
> 
> Great work John!
> Can you also add the flags to: https://github.com/Microsoft/openenclave/blob/master/cmake/maybe_build_using_clangw.cmake#L43
> 
> For building on windows, since cmake does not support multiple compilers/toolchain for different targets within the same project, we currently let cmake think that it is compiling enclaves using msvc and then override variables to cross-compile using clang. Therefore we need to specify the flags in two different places.

You got it! I'll send it through CI here to make sure it builds on Windows with this change :)  Thanks so much for your help, Anand!

bors try

Reply:
@anitagov I don't think there is an easy way to avoid duplication without putting some effort into refactoring our cmake infrastructure a bit. cmake has the notion of "current compiler" and we sometimes query to see if it supports a flag (spectre mitigation). I don't see a clean solution for avoiding the duplication.

Reply:
@anitagov Maybe we could define CLANG_FLAGS, GCC_FLAGS variables and reuse them?
In any case, feel free to take a stab if you have some free cycles. I don't think I can get to this any time soon.

Reply:
Yes, that should work. I can make this change with a different PR.